presto-tests/src/test/java/io/prestosql/execution/TestCoordinatorDynamicFiltering.java

```diff
@@ -0,0 +1,297 @@
+                    return new TestingHandleResolver();
+                }
+
+                @Override
+                public Connector create(String catalogName, Map<String, String> config, ConnectorContext context)
+                {
+                    return new TestConnector(metadata, Duration.valueOf("5s"), expectedDynamicFilter);
+                }
+            });
+        }
+    }
+
+    private static class TestConnector
+            implements Connector
+    {
+        private final ConnectorMetadata metadata;
+        private final Duration scanDuration;
+        private final AtomicReference<TupleDomain<ColumnHandle>> expectedDynamicFilter;
+
+        private TestConnector(ConnectorMetadata metadata, Duration scanDuration, AtomicReference<TupleDomain<ColumnHandle>> expectedDynamicFilter)
+        {
+            this.metadata = requireNonNull(metadata, "metadata is null");
+            this.scanDuration = requireNonNull(scanDuration, "scanDuration is null");
+            this.expectedDynamicFilter = requireNonNull(expectedDynamicFilter, "expectedDynamicFilter is null");
+        }
+
+        @Override
+        public ConnectorTransactionHandle beginTransaction(IsolationLevel isolationLevel, boolean readOnly)
+        {
+            return TestingTransactionHandle.create();
+        }
+
+        @Override
+        public ConnectorMetadata getMetadata(ConnectorTransactionHandle transactionHandle)
+        {
+            return metadata;
+        }
+
+        @Override
+        public ConnectorSplitManager getSplitManager()
+        {
+            return new ConnectorSplitManager()
+            {
+                @Override
+                public ConnectorSplitSource getSplits(
+                        ConnectorTransactionHandle transaction,
+                        ConnectorSession session,
+                        ConnectorTableHandle table,
+                        SplitSchedulingStrategy splitSchedulingStrategy,
+                        Supplier<TupleDomain<ColumnHandle>> dynamicFilter)
+                {
+                    long start = System.nanoTime();
+                    return new ConnectorSplitSource()
+                    {
+                        @Override
+                        public CompletableFuture<ConnectorSplitBatch> getNextBatch(ConnectorPartitionHandle partitionHandle, int maxSize)
+                        {
+                            return completedFuture(new ConnectorSplitBatch(
+                                    nCopies(maxSize, new EmptySplit(new CatalogName("test"))),
+                                    isFinished()));
+                        }
+
+                        @Override
+                        public void close()
+                        {
+                        }
+
+                        @Override
+                        public boolean isFinished()
+                        {
+                            if (Duration.nanosSince(start).compareTo(scanDuration) > 0) {
```

Comment:
this will cause each test to take 5 seconds.

Change the condition to:
```
if (dynamicFilter.get().equals(expectedDynamicFilter.get())) {
  return true;
} else if (Duration.nanosSince(start).compareTo(scanDuration) > 0) {
  throw new AssertionError(...))
}

// expected dynamic filter is not set yet
return false;
```

Reply:
Changed, also increased scanDuration to 10s to avoid any flakiness in tests from late arrival of DF.